tests/foreman/cli/test_remoteexecution.py

```diff
@@ -488,7 +497,9 @@ def test_positive_run_scheduled_job_template_by_ip(self, fixture_vmsetup, defaul
 
     @pytest.mark.tier3
     @pytest.mark.upgrade
-    def test_positive_run_receptor_installer(self, default_sat):
+    def test_positive_run_receptor_installer(
+        self, default_sat, subscribe_satellite, fixture_enable_receptor_repos
```

Comment:
subscribe_satellite is a `session-scoped` fixture and once it's used in the test it won't tear down for that test, instead, it will wait for the whole pytest session to complete. Do we ever consider the consequences of keeping it registered for the entire session that's initiated somewhere in the middle of a session from a test (like repos for the latest in-dev sat won't be available that dogfood does provide. There could be more which I cant calculate right now)? @SatelliteQE/robottelo-tier-1-reviewers  and @SatelliteQE/robottelo-tier-2-reviewers could jump in !

I would either turn that fixture into function scoped or create a new function scoped fixture to be used here! 

Thoughts @JacobCallahan @mshriver?

Reply:
I've discussed this for quite some time in chat because I don't like this approach either. It seems like using this fixture to do this is a commonly acceptable ground (after all, that's why it is there).
I still think, however, that the Satellite SHOULD be registered the whole time the automation runs against it.

If others think this fixture should be remade to be function-scoped and a teardown of `s-m unregister` should be added, I'm not against it but I'm not sure it's within scope of this PR - it could affect other tests (keeping it as it is has that potential, too, though).

Reply:
#9281 intends to solve the fixture issue